src/gbm/gbtree.cc

```diff
@@ -565,5 +604,51 @@ XGBOOST_REGISTER_GBM(Dart, "dart")
     GBTree* p = new Dart(base_margin);
     return p;
   });
+
+
+// FIXME: Better error handling.
+class TreeSelectRaw : public JsonReader {
+  size_t* pre_pos_ {nullptr};
+
+ public:
+  TreeSelectRaw(StringView str, size_t* pos) :
+      JsonReader{str, *pos}, pre_pos_{pos} {}
+
+  Json ParseRaw() {
+    SkipSpaces();
+    auto beg = cursor_.Pos();
+    while (true) {
+      char ch = GetNextChar();
+      while (ch != ']' && ch != -1) { ch = GetNextChar(); }
+      ch = GetNextNonSpaceChar();
```

Comment:
As far as I understand after line 622 we would have
ch == ']' or ch == -1
the first one is non white space. The second case indicates that the stream has been at the end.
Either way we do not need to skip non space chars. Or do I miss something here?